src/librustc_passes/stability.rs

```diff
@@ -36,6 +36,20 @@ enum AnnotationKind {
     Container,
 }
 
+/// Inheriting deprecations Nested items causes duplicate warnings.
+/// Inheriting the deprecation of `Foo<T>` onto the parameter `T`, would cause a duplicate warnings.
+#[derive(PartialEq, Copy, Clone)]
+enum InheritDeprecation {
+    Yes,
+    No,
+}
+
+impl InheritDeprecation {
+    fn yes(&self) -> bool {
+        *self == InheritDeprecation::Yes
```

Comment:
```suggestion
        matches!(self, InheritDeprecation::Yes)
```
Actually, using `matches!`, it might be cleaner simply to inline this and get rid of `PartialEq` on  `InheritDeprecation`.